src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java

```diff
@@ -224,22 +226,44 @@ public Integer getMaxComplaintsPerParticipant(Course course, Participant partici
     }
 
     /**
-     * This function checks whether the student is allowed to submit a complaint or not. Submitting a complaint is allowed within one week after the student received the result. If
-     * the result was submitted after the assessment due date or the assessment due date is not set, the completion date of the result is checked. If the result was submitted
-     * before the assessment due date, the assessment due date is checked, as the student can only see the result after the assessment due date.
+     * This function checks whether the student is allowed to submit a complaint / more feedback request or not.
+     * This is allowed within the corresponding number of days set in the course after the student received the result.
+     * If the result was submitted after the assessment due date or the assessment due date is not set, the completion date of the result is checked.
+     * If the result was submitted before the assessment due date, the assessment due date is checked, as the student can only see the result after the assessment due date.
      */
-    private boolean isTimeOfComplaintValid(Result result, Exercise exercise, Course course) {
+    private static void validateTimeOfComplaintOrRequestMoreFeedback(Result result, Exercise exercise, Course course, ComplaintType type) {
+        int maxDays = switch (type) {
+            case COMPLAINT -> course.getMaxComplaintTimeDays();
+            case MORE_FEEDBACK -> course.getMaxRequestMoreFeedbackTimeDays();
+        };
+
+        boolean isTimeValid;
         if (exercise.getAssessmentDueDate() == null || result.getCompletionDate().isAfter(exercise.getAssessmentDueDate())) {
-            return result.getCompletionDate().isAfter(ZonedDateTime.now().minusDays(course.getMaxComplaintTimeDays()));
+            isTimeValid = result.getCompletionDate().isAfter(ZonedDateTime.now().minusDays(maxDays));
+        }
+        else {
+            isTimeValid = exercise.getAssessmentDueDate().isAfter(ZonedDateTime.now().minusDays(maxDays));
+        }
+
+        if (!isTimeValid) {
+            StringBuilder message = new StringBuilder("You cannot ");
+            message.append(type == ComplaintType.COMPLAINT ? "submit a complaint" : "request more feedback");
+            message.append(" for a result that is older than ");
+            message.append(switch (maxDays) {
+                case 1 -> "one day";
+                case 7 -> "one week";
+                default -> maxDays + (maxDays % 7 == 0 ? " weeks" : " days");
```

Comment:
If maxDays is 14 this will result in "14 weeks" right?

Reply:
Yes, you are right, good catch. I will fix this.